src/filetree.rs

```diff
@@ -280,12 +280,63 @@ pub(crate) fn syncfs(d: &openat::Dir) -> Result<()> {
     rustix::fs::syncfs(d).map_err(Into::into)
 }
 
+fn copy_dir(src: &Path, dst: &Path) -> Result<()> {
+    let r = std::process::Command::new("cp")
+        .args(["-a"])
+        .arg(src)
+        .arg(dst)
+        .status()?;
+    if !r.success() {
+        anyhow::bail!("Failed to copy");
+    }
+    Ok(())
+}
+
+/// Convert path parent to temp, `fedora/foo` -> `fedora.tmp/foo`
 #[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
-fn tmpname_for_path<P: AsRef<Path>>(path: P) -> std::path::PathBuf {
+fn tmp_for_path<P: AsRef<Path>>(path: P) -> std::path::PathBuf {
     let path = path.as_ref();
-    let mut buf = path.file_name().expect("filename").to_os_string();
-    buf.push(TMP_PREFIX);
-    path.with_file_name(buf)
+    let mut parent = String::from("");
+    if let Some(p) = path.parent() {
+        parent = format!("{}.tmp", p.display());
+    }
+    let buf = path.file_name().expect("filename").to_os_string();
+    Path::new(parent.as_str()).join(buf)
+}
+
+/// Get diff based on sub dir
+#[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
+pub(crate) fn diff_with_prefix(diff: &FileTreeDiff, prefix: &str) -> Result<FileTreeDiff> {
+    let mut new_diff = FileTreeDiff {
+        additions: HashSet::new(),
+        removals: HashSet::new(),
+        changes: HashSet::new(),
+    };
+    for pathstr in diff.additions.iter() {
+        let path = Path::new(pathstr);
+        if let Some(parent) = path.parent() {
+            if parent.to_string_lossy().eq(prefix) {
```

Comment:
Let's avoid `to_string_lossy()`. Instead in general we can use https://docs.rs/camino/latest/camino/